src/Evolution/Systems/Cce/AnalyticSolutions/TeukolskyWave.cpp

```diff
@@ -0,0 +1,326 @@
+              pow<3>(extraction_radius_ * retarded_time) *
+              (5.0 * extraction_radius_ + retarded_time) +
+          6.0 * pow<6>(duration_) * extraction_radius_ *
+              (extraction_radius_ + retarded_time) *
+              (2.0 * extraction_radius_ + 7.0 * retarded_time) -
+          12.0 * pow<4>(duration_) * square(extraction_radius_) *
+              retarded_time *
+              (5.0 * square(extraction_radius_) +
+               4.0 * extraction_radius_ * retarded_time +
+               3.0 * square(retarded_time))) *
+         amplitude_ * exp(-square(retarded_time) / square(duration_)) /
+         (pow<10>(duration_) * pow<5>(extraction_radius_));
+}
+
+DataVector TeukolskyWave::sin_theta(const size_t l_max) noexcept {
+  DataVector sin_theta_result{
+      Spectral::Swsh::number_of_swsh_collocation_points(l_max)};
+  const auto& collocation = Spectral::Swsh::cached_collocation_metadata<
+      Spectral::Swsh::ComplexRepresentation::Interleaved>(l_max);
+  for (const auto& collocation_point : collocation) {
+    sin_theta_result[collocation_point.offset] = sin(collocation_point.theta);
+  }
+  return sin_theta_result;
+}
+
+DataVector TeukolskyWave::cos_theta(const size_t l_max) noexcept {
+  DataVector cos_theta_result{
+      Spectral::Swsh::number_of_swsh_collocation_points(l_max)};
+  const auto& collocation = Spectral::Swsh::cached_collocation_metadata<
+      Spectral::Swsh::ComplexRepresentation::Interleaved>(l_max);
+  for (const auto& collocation_point : collocation) {
+    cos_theta_result[collocation_point.offset] = cos(collocation_point.theta);
+  }
+  return cos_theta_result;
+}
+
+void TeukolskyWave::spherical_metric(
+    const gsl::not_null<
+        tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+        spherical_metric,
+    const size_t l_max, const double time) const noexcept {
+  const auto coefficient_a = pulse_profile_coefficient_a(time);
+  const auto coefficient_b = pulse_profile_coefficient_b(time);
+  const auto coefficient_c = pulse_profile_coefficient_c(time);
+
+  const auto local_sin_theta = sin_theta(l_max);
+  const auto local_cos_theta = cos_theta(l_max);
+
+  get<0, 0>(*spherical_metric) = -1.0;
+  get<0, 1>(*spherical_metric) = 0.0;
+  get<0, 2>(*spherical_metric) = 0.0;
+  get<0, 3>(*spherical_metric) = 0.0;
+
+  get<1, 3>(*spherical_metric) = 0.0;
+  get<2, 3>(*spherical_metric) = 0.0;
+  get<1, 1>(*spherical_metric) = 3.0 * square(local_cos_theta);
+  get<1, 2>(*spherical_metric) = -6.0 * local_sin_theta * local_cos_theta *
+                                 extraction_radius_ * coefficient_b;
+  get<2, 2>(*spherical_metric) =
+      (1.0 - coefficient_a + 3.0 * square(local_sin_theta) * coefficient_c) *
+      square(extraction_radius_);
+  get<3, 3>(*spherical_metric) =
+      (1.0 + (3.0 * square(local_sin_theta) - 1.0) * coefficient_a -
+       3.0 * square(local_sin_theta) * coefficient_c) *
+      square(extraction_radius_);
```

Comment:
missing factor sin^2

Reply:
This one is intentional -- the sin^2 theta is omitted because of the 'pfaffian' form